### DISCUSSION TITLE
Elastic Agents status show "Unhealthy" Elasticsearch connection failure

### DISCUSSION THREAD

**Discussion by Krapoh** [Security-Onion-Solutions/securityonion#10696] [2.4] [answered]
After we install SOC with version 2.4 and install an agent on the client-server on both Windows and Linux we found that the status of an agent always show Unhealthy "Elasticsearch connection failure" as picture below, may I know how to fix this issue?

Every we restart the agent service, the status is back to healthy then back to unhealthy again.

![Screenshot_1](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/f0cff59b-bab5-4f63-af98-d8f5613ba29c)

![Screenshot_2](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/d62f825e-73ca-47d9-ab96-d29d9e62b81c)

**Comment by defensivedepth** [comment, marked as answer]
This issue will be fixed in the next release.

**Reply by Krapoh** [reply]
May I know when it will release?

**Reply by dougburks** [reply]
2.4 Beta 4 is now available!
https://blog.securityonion.net/2023/07/security-onion-24-beta-4-release-now.html

**Reply by Krapoh** [reply]
OK, let me try on this version

**Reply by Krapoh** [reply]
Bad news, after installing the release version but the issue still occurs.
The status still shows "unhealthy" but no error about "Elasticsearch connection failure" inside.
What is the cause of this issue?

**Reply by defensivedepth** [reply]
If you click on the `Logs` tab, what errors are being reported?

**Reply by Krapoh** [reply]
There is no log as picture below and I have attached Diagnostics files zip to you

![Screenshot_1](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/efdc1065-c32b-4db1-a35d-f7ba006ac58b)


![Screenshot_2](https://github.com/Security-Onion-Solutions/securityonion/assets/136543416/1cc563c0-d6d5-405a-8687-50020c732ac7)


[elastic-agent-diagnostics-2023-07-17T03-56-43Z-00.zip](https://github.com/Security-Onion-Solutions/securityonion/files/12065959/elastic-agent-diagnostics-2023-07-17T03-56-43Z-00.zip)

**Reply by defensivedepth** [reply]
@Krapoh Couple things. The Elastic Agent diagnostics is a new feature that they just released - I have not confirmed that it automatically completely redacts sensitive information - please double-check this for yourself.

Have you changed the Outputs? Under Fleet --> Settings --> Outputs. The default should be set to `grid-logstash`. Looks like your instance is trying to connect to Elasticsearch on port 9200. We are using Logstash.

**Comment by charlespctech** [comment]
I am also running into this issue with the newest RC1 installation.

Endpoints showing as unhealthy with no windows logs being pushed over.

Do I need to go through these steps here to switch from the default "Elasticsearch" option to "Logstash" ?

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/7142801/99cc5cb9-904b-4d60-864c-cf69f770c5cd)

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/7142801/c60ede2c-8e50-442f-a965-28925fdb0b5d)

**Comment by charlespctech** [comment]
Actually I see why now in the documentation. I am running evaluation mode

In order to receive logs from the Elastic Agent, Security Onion must be running [Logstash](https://docs.securityonion.net/en/2.4/logstash.html#logstash). Evaluation Mode and Import Mode do not run [Logstash](https://docs.securityonion.net/en/2.4/logstash.html#logstash), so you’ll need Standalone or a full Distributed Deployment. In a Distributed Deployment, forward nodes do not run [Logstash](https://docs.securityonion.net/en/2.4/logstash.html#logstash), so you’ll need to configure agents to send to your manager or receiver nodes. For more information, please see the [Architecture](https://docs.securityonion.net/en/2.4/architecture.html#architecture) section.
